src/main.cpp

```diff
@@ -59,22 +59,29 @@ std::string make_response(const json& response)
     return header + content.dump(4);
 }
 
+static bool ends_with(const std::string haystack, const std::string needle) {
+    if (needle.length() > haystack.length())
+        return false;
+    return haystack.compare(haystack.length() - needle.length(), needle.length(), needle) == 0;
+}
+
 EShLanguage find_language(const std::string& name)
```

Comment:
This function has become somewhat complicated - Please set a good example for further contributors now and start documenting this function.